src/runtime/propensity-server.js

```diff
@@ -14,7 +14,34 @@
  * limitations under the License.
  */
 import {Xhr} from '../utils/xhr';
-import * as ServiceUrl from './services';
+import {adsUrl} from './services';
+import {Event} from '../api/propensity-api';
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import {isObject,isBoolean} from '../utils/types';
+import {ExperimentFlags} from './experiment-flags';
+import {isExperimentOn} from './experiments';
+
+/** @private @const {!Object<number,string>} */
+const AnalyticsEventToPropensityEvent = {
+  [AnalyticsEvent.UNKNOWN]: null,
+  [AnalyticsEvent.IMPRESSION_PAYWALL]: Event.IMPRESSION_PAYWALL,
+  [AnalyticsEvent.IMPRESSION_AD]: Event.IMPRESSION_AD,
+  [AnalyticsEvent.IMPRESSION_OFFERS]: Event.IMPRESSION_OFFERS,
+  [AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON]: null,
+  [AnalyticsEvent.IMPRESSION_SMARTBOX]: null,
+  [AnalyticsEvent.ACTION_SUBSCRIBE]: null,
```

Comment:
AnalytisEvent.ACTION_SUBSCRIBE maps to Event.ACTION_PAYMENT_FLOW_STARTED

Reply:
There is also an AnalyticsEvent.ACTION_PAY_FLOW_STARTED.

Reply:
I see that ACTION_SUBSCRIBE is used in 1 place in the codebase: it might be preferable to replace it with ACTION_PAYMENT_FLOW_STARTED if that is in fact what it means.  It's a much better enum value

Reply:
my bad: This is the right mapping.
  [AnalyticsEvent.ACTION_PAYMENT_FLOW_STARTED]:
      Event.ACTION_PAYMENT_FLOW_STARTED,

ACTION_SUBSCRIBE tracks click on subscribe button. There's no equivalent of that in Propensity world. Please ignore this whole set.